common/persistence/sql/sqlClusterMetadataManager.go

```diff
@@ -97,10 +96,13 @@ func (s *sqlClusterMetadataManager) GetImmutableClusterMetadata() (*p.InternalGe
 }
 
 func (s *sqlClusterMetadataManager) GetClusterMembers(request *p.GetClusterMembersRequest) (*p.GetClusterMembersResponse, error) {
-	pageToken := uint64(0)
-	if len(request.NextPageToken) > 0 {
-		pageToken = binary.LittleEndian.Uint64(request.NextPageToken)
+	var lastSeenHostId []byte
+	if len(request.NextPageToken) == 16 {
+		lastSeenHostId = request.NextPageToken
```

Comment:
these UUIDs are going to be completely random, right? Technically that means that during paging, any newly added hosts that show up might have a UUID before the current page token and not be returned. Not really a big deal (new hosts can show up even after I finish the read), but just want to make sure this is intended and there is no weirdness that this behavior is random.

If we are actually worried about this, then we can also include the "last updated time" in the pageToken and ignore entries on subsequent page retrievals that were added after the time at which the first page was queried

Reply:
We do lose the guarantee that we had previously that we would include all newly added hosts even if added in between pages because of the global ordering we had. This is nice, but shouldn't be required, especially given the retry logic you added recently. 

Even though it's random, this is an optimization in a way at this point (given above though its okay if we left them out as well). A better model could be changing the sorting order to paging by Session_start as this would roughly model the previous behaviors. However this introduces similar missing results scenarios during paging due to introducing clock skew in ordering or if there is enough time collisions due to poor time granularities in the sql store. I went down this path a bit and just the different time granularities are a headache.